mne/annotations.py

```diff
@@ -214,6 +214,37 @@ def __iadd__(self, other):
                                                  other.orig_time))
         return self.append(other.onset, other.duration, other.description)
 
+    def __iter__(self):
+        for idx in range(len(self.onset)):
+            yield (self.onset[idx], self.duration[idx], self.description[idx])
+
+    def __getitem__(self, key):
+        """Propagate indexing and slicing to the underlying numpy structure."""
+        try:
+            if isinstance(key, slice) or isinstance(key, list):
```

Comment:
I would do:

onset=self.onset[key]

and then test that if onset.ndim = 1 then return a dict. It avoids the try complication

Reply:
The complication of the trys is just to raise `TypeError` instead of `IndexError` when due. (Following the standard recommendations)

and the ifs (to check it before `self.onset[key]` and not after) is to allow getting one element annotations. 

as in here 
https://github.com/mne-tools/mne-python/pull/5800/files#diff-0fbcacc4dac884a4e8bc277be5eaa2efR828

I think that this should hold:

```py
assert isinstance(annotations(slice(0, None, None)), Annotations)
```

Reply:
you should test if key is iterable as if you pass a tuple or a numpy array of ints it should work too.